lib/Onyx.js

```diff
@@ -1025,13 +1025,18 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * You may pass in additionalDefaults to set/preserve onyx data that you want to keep
+ * after clearing.
+ *
+ * @param {Object} additionalDefaults additional default values for ONYXKEYS keys
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(additionalDefaults = {}) {
+    const keyStatesToPreserve = fastMerge(defaultKeyStates, additionalDefaults);
```

Comment:
```suggestion
function clear(keyStatesToPreserve = {}) {
    const defaultValues = fastMerge(defaultKeyStates, keyStatesToPreserve);
```

I think this is clearer naming since the default keys are not necessarily things we are preserving.

Reply:
I don't know if we want to address this here, but I find the existing code below this strange. We are clearing the storage entirely... won't this remove the default and preserved states from the device storage? If we want to "preserve" something would a more correct approach be to just skip it's deletion?

Reply:
Yes, I think that would be a better approach. Kind of been thinking about switching it over to this too as I've been looking at the /App side of things.

Reply:
Maybe the idea with keeping the defaultKeyStates in the cache only was that the defaultKeyStates are not saved to storage when Onyx is initialized?

I'm fine with switching to an approach where default keys are not cleared from the cache or storage.

Reply:
Sounds good - gonna push that change I tested it with the linked PR and it works just the same :+1:

Reply:
> Sounds good - gonna push that change I tested it with the linked PR and it works just the same 👍

Uhhh... nvm I don't think I was quite understanding what I was doing 🤦‍♂️ 

I _think_ I understand this a bit clearer now...

> I'm fine with switching to an approach where default keys are not cleared from the cache or storage.

So I'm guessing this would necessitate removing each value one by one?